src/test/java/core/basesyntax/service/impl/FileReaderCsvTest.java

```diff
@@ -0,0 +1,95 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.exception.InvalidDataException;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.storage.Storage;
+import core.basesyntax.strategy.OperationHandler;
+import core.basesyntax.strategy.impl.BalanceHandler;
+import core.basesyntax.strategy.impl.PurchaseHandler;
+import core.basesyntax.strategy.impl.ReturnHandler;
+import core.basesyntax.strategy.impl.SupplyHandler;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+public class FileReaderCsvTest {
+    private static final String DEFAULT_VALID_FILE = "src/main/resources/example.csv";
+    private static final String INVALID_PATH_TO_FILE = "src/main/resources/example3.csv";
+    private static final String INVALID_QUANTITY_IN_FILE = "src/main/resources/example2.csv";
+    private static final String NULL_QUANTITY_IN_FILE = "src/main/resources/"
+            + "exampleWithNullQuantity.csv";
+    private static final String NULL_FRUIT_IN_FILE = "src/main/resources/exampleWithNullFruit.csv";
+    private FileReaderCsv fileReaderCsv = new FileReaderCsv();
+    private FruitTransactionParserImpl fruitTransactionParser = new FruitTransactionParserImpl();
+    private ReportGenerator reportGenerator = new ReportGenerator();
+
+    @Test
+    void read_validPath_Ok() {
+        List<String> actual = fileReaderCsv.read(DEFAULT_VALID_FILE);
+        Assertions.assertEquals(actual, fileReaderCsv.read(DEFAULT_VALID_FILE));
+    }
+
+    @Test
+    void read_invalidPath_notOk() {
+        Throwable exception = Assertions.assertThrows(InvalidDataException.class, () -> {
+            fileReaderCsv.read(INVALID_PATH_TO_FILE);
+        });
+        Assertions.assertEquals("Can't read file " + INVALID_PATH_TO_FILE, exception.getMessage());
+    }
+
+    @Test
+    void parse_validData_Ok() {
+        List<FruitTransaction> actual = fruitTransactionParser.parse(fileReaderCsv
+                .read(DEFAULT_VALID_FILE));
+        List<FruitTransaction> expected = new ArrayList<>();
+        Assertions.assertSame(expected.getClass(), actual.getClass());
+    }
+
+    @Test
+    void parse_invalidQuantity_notOk() {
+        Throwable exception = Assertions.assertThrows(InvalidDataException.class, () -> {
+            fruitTransactionParser.parse(fileReaderCsv.read(INVALID_QUANTITY_IN_FILE));
+        });
+        Assertions.assertEquals("Input quantity is not valid", exception.getMessage());
```

Comment:
You can simplify the code like this:
```suggestion
      assertThrows(InvalidDataException.class, () ->
        fruitTransactionParser.parse(fileReaderCsv.read(INVALID_QUANTITY_IN_FILE)),
        "Input quantity is not valid");`
```
It`s up to you)